fabric-convention-tags-v2/src/client/java/net/fabricmc/fabric/impl/tag/convention/ConventionLogWarningsClient.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.tag.convention;
+
+import java.util.List;
+import java.util.Locale;
+
+import it.unimi.dsi.fastutil.objects.ObjectArrayList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.minecraft.client.resource.language.I18n;
+import net.minecraft.item.Item;
+import net.minecraft.registry.Registry;
+import net.minecraft.registry.RegistryKeys;
+import net.minecraft.registry.tag.TagKey;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.api.ClientModInitializer;
+import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents;
+import net.fabricmc.loader.api.FabricLoader;
+
+public class ConventionLogWarningsClient implements ClientModInitializer {
+	private static final Logger LOGGER = LoggerFactory.getLogger(ConventionLogWarningsClient.class);
+
+	/**
+	 * A config option mainly for developers.
+	 * Logs out modded item tags that do not have translations when running on integrated server.
+	 * Defaults to SILENCED.
+	 */
+	private static final LogWarningMode LOG_UNTRANSLATED_WARNING_MODE = setupLogWarningModeProperty();
+
+	private static LogWarningMode setupLogWarningModeProperty() {
+		String property = System.getProperty("fabric-tag-conventions-v2.missingTagTranslationWarning", LogWarningMode.SILENCED.name()).toUpperCase(Locale.ROOT);
+
+		try {
+			return LogWarningMode.valueOf(property);
+		} catch (Exception e) {
+			LOGGER.error("Unknown entry `{}` for property `fabric-tag-conventions-v2.missingTagTranslationWarning`.", property);
+			return LogWarningMode.SILENCED;
+		}
+	}
+
+	private enum LogWarningMode {
+		SILENCED,
+		DEV_SHORT,
+		DEV_VERBOSE
+	}
+
+	@Override
+	public void onInitializeClient() {
+		if (LOG_UNTRANSLATED_WARNING_MODE != LogWarningMode.SILENCED) {
+			setupUntranslatedItemTagWarning();
+		}
+	}
+
+	private static void setupUntranslatedItemTagWarning() {
+		// Log missing item tag translations only in development environment and not running dedicated server.
+		ServerLifecycleEvents.SERVER_STARTED.register(server -> {
+			boolean isConfigSetToDev =
+					LOG_UNTRANSLATED_WARNING_MODE == LogWarningMode.DEV_SHORT
+							|| LOG_UNTRANSLATED_WARNING_MODE == LogWarningMode.DEV_VERBOSE;
+
+			if (FabricLoader.getInstance().isDevelopmentEnvironment() != isConfigSetToDev) {
```

Comment:
Instead of this contrived formula, just check if it is dev. The only config not set to dev is SILENCE, but in that case this appears to be a dead code.